src/codec.rs

```diff
@@ -221,7 +221,11 @@ pub trait Output: Sized {
 	}
 
 	/// Write encoding of given value to the output.
-	fn push<V: Encode + ?Sized>(&mut self, value: &V) {
+	///
+	/// NOTE: this is only available when `Self` is `Sized`, thus it is not available in
+	/// `Encode::encode_to`. If you need it when `Self` is not `Sized` consider using
+	/// `Encode::encode_to` directly.
+	fn push<V: Encode + ?Sized>(&mut self, value: &V) where Self: Sized {
```

Comment:
Isn't the trick here to just take `value: &dyn Encode` to make this available on the trait? From my side we could also drop this function. 

Or there is another trick, we can move this into some own trait and impl this trait for `dyn Output` :D

Reply:
dyn Encode doesn't work as Encode cannot be made into trait object

But implementing a trait PushOutput to dyn Output works, but then it is no longer implemented on not-dyn Output (I think),
I guess we can have both,

Anyway I removed it because it brings no help and might confuse some people